Orange/widgets/data/owmergedata.py

```diff
@@ -566,6 +557,29 @@ def _join_table_by_indices(self, reduced_extra, lefti, righti, rightu):
 
         return table
 
+    def _domain_rename_duplicates(self, domain):
+        """Check for duplicate variable names in domain. If any, rename
+        the variables, by replacing them with new ones (names are
+        appended a number). """
+        attrs, cvars, metas = [], [], []
+        n_attrs, n_cvars, n_metas = (len(domain.attributes),
+                                     len(domain.class_vars), len(domain.metas))
+        lists = [attrs] * n_attrs + [cvars] * n_cvars + [metas] * n_metas
+
+        variables = domain.variables + domain.metas
+        proposed_names = [m.name for m in variables]
+        unique_names = get_unique_names_duplicates(proposed_names)
+        duplicates = set()
+        for p_name, u_name, var, c in zip(proposed_names, unique_names,
+                                          variables, lists):
+            if p_name != u_name:
+                duplicates.add(p_name)
+                var = var.rename(u_name)
```

Comment:
... and then this would be `var = var.copy(name=u_name)`.

Reply:
I'd prefer `var = var.copy(compute_value=var.compute_value, name=u_name)`.

Speaking of properties we want to keep. Why not copying `color` as well?

Reply:
| `var = var.copy(compute_value=var.compute_value, name=u_name)`

OK, if you wish. I view `copy` as a function that copies the variable, but because its attributes are constant, you can also override some by giving them as keyword arguments. Much like `_replace` in `namedtuple` (which is also misleading: for some time I wondered how can it modify a tuple!). This is why I wouldn't pass a value that I am not changing (`compute_value`). If you pass `compute_value`, why not also `values` and `number_of_decimals`?

I think that `_colors` doesn't need to be copied because it is also saved in `attributes`. I hope this is true.

Reply:
On the other hand. `compute_value` could be considered same as `name`: `if compute_value is None: compute_value = self.compute_value`.

Reply:
Yes. On the second point. Anything you do not explicitly change is (should be) copied.